tensorflow/python/keras/layers/preprocessing/text_vectorization.py

```diff
@@ -537,31 +537,25 @@ def call(self, inputs):
     # If we're not doing any output processing, return right away.
     if self._output_mode is None:
       return inputs
+
     lookup_data = self._index_lookup_layer(inputs)
     if self._output_mode == INT:
-      # Once we have the dense tensor, we can return it if we weren't given a
-      # fixed output sequence length. If we were, though, we have to dynamically
-      # choose whether to pad or trim it based on each tensor.
 
-      # We need to convert to dense if we have a ragged tensor.
-      if tf_utils.is_ragged(lookup_data):
-        dense_data = lookup_data.to_tensor(default_value=0)
-      else:
-        dense_data = lookup_data
+      # Trim the output (magic of None slicing)
```

Comment:
Let's make the "magic" explicit

`# Trim the output. Will do nothing if _output_sequence_length is null.`

Or something like that

Reply:
Thanks, @mattdangerw 
WE added some unit tests ;)